### PR TITLE
Merge release 20.7 code freeze into `trunk`

### PR DESCRIPTION
**Pull request by mokagio**
- [x] Switch to production Pods
- [x] New version header in `RELEASE-NOTE.txt` 
- [x] `Localizable.strings` updated 
- [x] `release_notes.txt` updated with notes from `RELEASE-NOTE.txt` for current version 
- [x] Version update in `.xcconfig`

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -222,7 +222,7 @@ abstract_target 'Apps' do
 
   pod 'Gridicons', '~> 1.1.0'
 
-  pod 'WordPressAuthenticator', '~> 2.1'
+  pod 'WordPressAuthenticator', '~> 2.3.0'
```
Comment (mokagio) [inline Podfile:225]: This is the rule necessary to not automatically to `2.4.0`, which introduced a breaking change.

See
https://buildkite.com/automattic/wordpress-ios/builds/9808#01830b2b-76f3-4fe1-b385-52e09f417c3c:

```
WordPress/Classes/ViewRelated/NUX/WordPressAuthenticationManager.swift:228:1: type 'WordPressAuthenticationManager' does not conform to protocol 'WordPressAuthenticatorDelegate'
```

File: config/Version.public.xcconfig [hunk 2]
```diff
@@ -1,4 +1,4 @@
-VERSION_SHORT=20.6
+VERSION_SHORT=20.7
 
 // Public long version example: VERSION_LONG=9.9.0.0
-VERSION_LONG=20.6.0.3
+VERSION_LONG=20.7.0.0
```
Comment (mokagio) [inline config/Version.public.xcconfig:4]: While I run multiple release build jobs in CI, only the last one succeeded, so there was no need to further bump the build number.

File: WordPress/Resources/release_notes.txt [hunk 3]
```diff
@@ -1,3 +0,0 @@
-Stay warm, friends—the pages list will no longer freeze when a page’s featured image is a GIF.
-
-File Download stats aren’t supported on self-hosted Jetpack sites, so you won’t see that section in your Stats section anymore.
```
Comment (mokagio) [inline WordPress/Resources/release_notes.txt:3]: I reached out on our internal Slack asking the iOS devs if there's anything they'd like to highlight for this release.

**Comment by AliSoftware** [comment]
TestFlight build `20.7.0.0` of Jetpack iOS got rejected by Apple (ref: p1662374430385989-slack-CC7L49W13) 😞 

<details><summary>Rejection email</summary>

> Dear Developer,
> 
> We identified one or more issues with a recent submission for TestFlight review for your app, "Jetpack – Website Builder" 20.7 (20.7.0.0). Please correct the following issues, then upload again.
> 
> ITMS-90433: Invalid Swift Support - The file libswift_Concurrency.dylib doesn't have the correct code signature. Make sure you're using the correct signature, rebuild your app using the current public (GM) version of Xcode, and resubmit it. Don't just modify the code signature of libswift_Concurrency.dylib.
> 
> Best regards,
> 
> The App Store Team
</details>

**Review by crazytonyli — APPROVED** [review]

**Comment by mokagio** [comment]
Closing in favor of https://github.com/wordpress-mobile/WordPress-iOS/pull/19288, which addresses the `RELEASE-NOTES.txt` conflict.

**Pull request closed without merging**